### PR TITLE
adding types package

### PR DESCRIPTION
**Pull request by cfortuner**
I was able to get this to work by creating a @types/window.ai package.

In this implementation,

in the `web` app, typescript is able to find the types for window.ai after installing @types/window.ai with no other changes required.

---

I also tried to get this to work by just creating a package `windowai` and including the types in the package.json as 
`types: 'index.d.ts'`

But in this case, I wasn't able to get typescript to find the types without requiring the user to modify their tsconfig.json

--- 

I think there *should* be a way to publish a npm package, that includes types that are found by ts without requiring any other changes, but i'm not sure how to do this yet.

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -5,6 +5,7 @@
 # dependencies
 .turbo
 /node_modules
+**node_modules
```
Comment (alexanderatallah) [inline .gitignore:8]: why is this needed? shouldn't each sub-package have its own gitignore?

File: apps/web/components/Chat.tsx [hunk 2]
```diff
@@ -1,19 +1,18 @@
...
-import { DISCORD_URL, DOWNLOAD_URL } from "./common"
-import * as ai from "window-ai"
+import { useEffect, useRef, useState } from "react";
+import { Button } from "./Button";
+import { type ChatGPTMessage, ChatLine, LoadingChatLine } from "./ChatLine";
+import { useCookies } from "react-cookie";
+import { DISCORD_URL, DOWNLOAD_URL } from "./common";
 
-const COOKIE_NAME = "nextjs-example-ai-chat-gpt3"
+const COOKIE_NAME = "nextjs-example-ai-chat-gpt3";
```
Comment (alexanderatallah) [inline apps/web/components/Chat.tsx:7]: should we standardize to no semicolons and put eslint config in a package?

File: packages/types/index.d.ts [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+
+export type Output =
+  | {
+      text: string;
+    }
+  | {
+      message: ChatMessage;
+    };
+
+type LLM =
```
Comment (alexanderatallah) [inline packages/types/index.d.ts:19]: is there a way to do this with an enum (even though it's technically an object type)? wondering if we should make it easier for people to enumerate over the ModelIDs, which you can't do this way.

The best types are in `extension/src/public-interface.ts`

File: packages/types/index.d.ts [hunk 4]
```diff
@@ -0,0 +1,57 @@
...
+  | "openai/gpt3.5"
+  | "together/gpt-neoxt-20B"
+  | "cohere/xlarge"
+  | "local";
+
+export interface CompletionOptions {
+  onStreamResult?: (result: Output | null, error: string | null) => unknown;
```
Comment (alexanderatallah) [inline packages/types/index.d.ts:26]: there are lots of comments on these options in the `public-interface.ts` file - wondering if we can preserve those so VSCode shows them when you're coding?

File: packages/types/package.json [hunk 5]
```diff
@@ -1,10 +1,11 @@
...
-  "name": "window-ai",
-  "main": "index.d.ts",
+  "name": "@types/window.ai",
   "types": "index.d.ts",
   "version": "1.0.0",
-  "description": "Types for window.ai",
-  "scripts": {},
+  "description": "Window.ai API",
+  "scripts": {
+    "publish": "npm publish --access public"
```
Comment (alexanderatallah) [inline packages/types/package.json:7]: pnpm has its own `pnpm publish` script for this i think

**Review by alexanderatallah — CHANGES_REQUESTED** [review]
added comments

File: apps/web/components/Chat.tsx (hunk 2 above)
Comment (cfortuner) [reply apps/web/components/Chat.tsx:7]: yep just haven't gotten around this yet!

Comment (cfortuner) [reply apps/web/components/Chat.tsx]: adding prettier and eslint to the root

File: packages/types/index.d.ts (hunk 3 above)
Comment (cfortuner) [reply packages/types/index.d.ts]: yeah I think we can do this with our own lib or rely on a community contribution

File: packages/types/index.d.ts [hunk 6]
```diff
@@ -0,0 +1,95 @@
...
+  }
+
+  interface Window {
+    ai: WindowAI
+  }
+}
+
+export {} // This is necessary for module augmentation
+
+window.ai.getCompletion
```
Comment (alexanderatallah) [inline packages/types/index.d.ts:95]: ha what's this doing here

**Review by alexanderatallah — APPROVED** [review]
approved pending one small comment!
